### PR TITLE
Add fixed DNS code

### PR DESCRIPTION
**Pull request by axxName**
I corrected some stupid mistakes in the code (ext-dns.c)
Please test again.

And with nonexistent DNS server:
kadnode --dns-server 192.168.99.99 --dns-timeout 1

### CODE REVIEW

**Comment by mwarning** [comment]
@axxName I have added some dns proxy code that seem to work. It does not block KadNode and is a bit simpler (no timeout needed) even.
Maybe you can give it a try and create a merge request for the rest of the windows setup.
I would be happy to merge it.

File: windows/dns_reset.bat [hunk 1]
```diff
@@ -15,8 +15,17 @@ SET adapterName=!adapterName:~17!
 REM Removes the colon from the end of the adapter name
 SET adapterName=!adapterName:~0,-1!
 
-netsh interface ip set address "!adapterName!" dhcp
-netsh interface ip delete dnsserver "!adapterName!" all
+set skip=0
+
+ECHO.!adapterName!| FIND /I "vpn">Nul && ( set skip=1 )
+ECHO.!adapterName!| FIND /I "virtual">Nul && ( set skip=1 )
+
```
Comment (mwarning) [inline windows/dns_reset.bat]: Good, I have not thought about vpn and virtual interfaces in this context. :-)
Is FIND case insensitive?

Comment (axxName) [inline windows/dns_reset.bat]: A key   "/I  "  -  case insensitive

Comment (axxName) [inline windows/dns_reset.bat]: About timeout and other code 
It was an attempt to use multiple DNS servers if one is unavailable.

OK. It is Your project ;)

Comment (mwarning) [inline windows/dns_reset.bat]: @axxName it was just much more easier to implement the code for one upstream DNS server. /etc/resolv.conf uses at most two DNS servers anyway.
About the timeout, it is not necessary the way it is implemented right now. The timeout is dictated by the amount of requests per seconds.

Comment (mwarning) [inline windows/dns_reset.bat]: sorry about not merging your code, but I've checked it out and modified it until it became the code that was merged.

Comment (axxName) [inline windows/dns_reset.bat]: It's OK.
I am new to C programming.
And I think Your code is better.
The main goal accomplished. Kadnode supports all DNS queries.

I will make a new pull request later.

**Pull request closed without merging**